test/integration/integration_admin_test.cc

```diff
@@ -53,6 +53,55 @@ TEST_P(IntegrationAdminTest, HealthCheck) {
   EXPECT_EQ("200", response->headers().Status()->value().getStringView());
 }
 
+TEST_P(IntegrationAdminTest, HealthCheckWithoutServerStats) {
+  envoy::config::metrics::v2::StatsMatcher stats_matcher;
+  stats_matcher.mutable_exclusion_list()->add_patterns()->set_prefix("server.");
+  initialize(stats_matcher);
+
+  BufferingStreamDecoderPtr response = IntegrationUtil::makeSingleRequest(
+      lookupPort("http"), "POST", "/healthcheck", "", downstreamProtocol(), version_);
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().Status()->value().getStringView());
+
+  response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "GET", "/stats", "",
+                                                downstreamProtocol(), version_);
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().Status()->value().getStringView());
+  EXPECT_THAT(response->body(), testing::Not(testing::HasSubstr("server.")));
+
+  response = IntegrationUtil::makeSingleRequest(lookupPort("admin"), "POST", "/healthcheck/fail",
+                                                "", downstreamProtocol(), version_);
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("200", response->headers().Status()->value().getStringView());
+
+  response = IntegrationUtil::makeSingleRequest(lookupPort("http"), "GET", "/healthcheck", "",
+                                                downstreamProtocol(), version_);
+  EXPECT_TRUE(response->complete());
+  EXPECT_EQ("503", response->headers().Status()->value().getStringView());
```

Comment:
I wasn't expecting the healthcheck call to return a "503" -- is that how it works currently?

Reply:
Yes, I believe that's what is returned currently.